### PR TITLE
buildGoModule: Add bazaar support

### PR DESCRIPTION
**Pull request by trcm**
###### Motivation for this change

I had some failing build for go modules due to the dependencies using bazaar rather than git.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by worldofpeace** [comment]
I've requested @kalbasit to review your PR @trcm because he maintains this framework.


Other than that, please merge instead of rebasing per the contribution guidelines
and we need to resolve the conflicts.

File: pkgs/development/go-modules/generic/default.nix [hunk 1]
```diff
@@ -33,10 +33,16 @@ let
 
   removeExpr = refs: ''remove-references-to ${lib.concatMapStrings (ref: " -t ${ref}") refs}'';
 
+  bazaarNoCertValidation =
+    runCommandNoCC "bzr-no-cert-validation" {
+      inherit bazaar;
+      buildInputs = [ makeWrapper ];
+    } "makeWrapper $bazaar/bin/bzr $out/bin/bzr --add-flags -Ossl.cert_reqs=none";
```
Comment (kalbasit) [inline pkgs/development/go-modules/generic/default.nix]: Is there a way to provide the cacert instead of disabling TLS altogether?

**Review by kalbasit — CHANGES_REQUESTED** [review]

**Comment by Infinisil** [comment]
What's the status of this?

**Comment by kalbasit** [comment]
I'd say it's stalled. @trcm are you interested in getting this off to the finish line?

**Comment by trcm** [comment]
Yeah sorry, I'll circle back around and have a look at this

**Pull request closed without merging**